routing/base/astar_algorithm.hpp

```diff
@@ -315,16 +323,22 @@ template <typename TGraph>
 typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPath(
     TGraphType & graph, TVertexType const & startVertex, TVertexType const & finalVertex,
     RoutingResult<TVertexType, TWeightType> & result, my::Cancellable const & cancellable,
-    TOnVisitedVertexCallback onVisitedVertexCallback) const
+    OnVisitedVertexCallback onVisitedVertexCallback, CheckWeightCallback checkWeightCallback) const
 {
   result.Clear();
   if (nullptr == onVisitedVertexCallback)
     onVisitedVertexCallback = [](TVertexType const &, TVertexType const &) {};
 
+  if (checkWeightCallback == nullptr)
+    checkWeightCallback = [](TWeightType const & /* weight */) { return true; };
+
   Context context;
   PeriodicPollCancellable periodicCancellable(cancellable);
   Result resultCode = Result::NoPath;
 
+  auto const piS = graph.HeuristicCostEstimate(startVertex, finalVertex);
+  CHECK_EQUAL(graph.HeuristicCostEstimate(finalVertex, finalVertex), kZeroDistance, ("Heuristic is not consistent."));
```

Comment:
It may be not exactly zero, but very close. Please, compare with kEpsilon.

Reply:
done